src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs

```diff
@@ -159,13 +156,14 @@ private bool Locked
         }
     }
 
-    private string Name
+    private string? Name
     {
-        get => Component.Site.Name;
+        get => Component.Site?.Name;
         set
         {
             // Don't do anything here during loading, if a refactor changed it we don't want to do anything.
-            if (GetService(typeof(IDesignerHost)) is not IDesignerHost host || (host is not null && !host.Loading))
+            if ((TryGetService(out IDesignerHost? host) || (host is not null && !host.Loading))
```

Comment:
Looks like we missed a `!` this might be why some tests are failing.
```suggestion
            if (!TryGetService(out IDesignerHost? host) || (host is not null && !host.Loading))
```